### PR TITLE
Docker connection

### PR DESCRIPTION
**Pull request by aviralsrivastava01**
# Description

Self Signed Certificate for private repos and Insecure registries allow for docker registry.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Docker registry Create, Get, Update
- [x] Ci Trigger With All the cases [Secure/SecureWithCert/Insecure]

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

**Review by vikramdevtron — DISMISSED** [review]

File: api/restHandler/DockerRegRestHandler.go [hunk 1]
```diff
@@ -105,7 +107,11 @@ func (impl DockerRegRestHandlerImpl) SaveDockerRegistryConfig(w http.ResponseWri
 		writeJsonResp(w, err, nil, http.StatusInternalServerError)
 		return
 	}
-	writeJsonResp(w, err, res, http.StatusOK)
+	if (res.Connection=="secure-with-cert" && res.Cert == "") || (res.Connection!="secure-with-cert" && res.Cert != "") {
```
Comment (vikramdevtron) [inline api/restHandler/DockerRegRestHandler.go:110]: use constants instead flat string "secure-with-cert"

File: api/restHandler/DockerRegRestHandler.go [hunk 2]
```diff
@@ -210,7 +216,11 @@ func (impl DockerRegRestHandlerImpl) UpdateDockerRegistryConfig(w http.ResponseW
...
```
Comment (vikramdevtron) [inline api/restHandler/DockerRegRestHandler.go:219]: use constants instead flat string "secure-with-cert"

File: api/restHandler/DockerRegRestHandler.go (hunk 1 above)
Comment (vikramdevtron) [inline api/restHandler/DockerRegRestHandler.go:110]: validation check add just before making db call not after that

File: api/restHandler/DockerRegRestHandler.go [hunk 3]
```diff
@@ -105,7 +107,11 @@ func (impl DockerRegRestHandlerImpl) SaveDockerRegistryConfig(w http.ResponseWri
 		writeJsonResp(w, err, nil, http.StatusInternalServerError)
 		return
 	}
-	writeJsonResp(w, err, res, http.StatusOK)
+	if (res.Connection==secureWithCert && res.Cert == "") || (res.Connection!=secureWithCert && res.Cert != "") {
```
Comment (vikramdevtron) [inline api/restHandler/DockerRegRestHandler.go:110]: this check has to be before service call

File: api/restHandler/DockerRegRestHandler.go [hunk 4]
```diff
@@ -210,7 +216,12 @@ func (impl DockerRegRestHandlerImpl) UpdateDockerRegistryConfig(w http.ResponseW
...
```
Comment (vikramdevtron) [inline api/restHandler/DockerRegRestHandler.go:219]: this check has to be before service call

**Review by vikramdevtron — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**